src/targets/awsLambda.ts

```diff
@@ -0,0 +1,187 @@
+ */
+export const defaultLayerName = 'SentryNodeServerlessSDK';
+
+/**
+ * Extracts the AWS Lambda layer name from the environment variables. If no
+ * environment variable is found, the default name is used.
+ */
+export function getAwsLayerName(): string {
+  if (!process.env.AWS_LAYER_NAME) {
+    return defaultLayerName;
+  } else {
+    return process.env.AWS_LAYER_NAME;
+  }
+}
+
+/**
+ * Target responsible for uploading files to AWS Lambda.
+ */
+export class AwsLambdaTarget extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'aws-lambda';
+  /** Target options */
+  public readonly awsLambdaConfig: AwsLambdaTargetOptions;
+  /** Name of the layer to be published */
+  static layerName: string;
+
+  public constructor(
+    config: TargetConfig,
+    artifactProvider: BaseArtifactProvider
+  ) {
+    super(config, artifactProvider);
+    this.awsLambdaConfig = this.getAwsLambdaConfig();
+    AwsLambdaTarget.layerName = getAwsLayerName();
+  }
+
+  /**
+   * Extracts AWS Lambda target options from the environment.
+   */
+  protected getAwsLambdaConfig(): AwsLambdaTargetOptions {
+    if (!process.env.AWS_ACCESS_KEY_ID || !process.env.AWS_SECRET_ACCESS_KEY) {
+      throw new ConfigurationError(
+        `Cannot perform AWS Lambda release: missing credentials.
+        Please use AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables.`
+      );
+    }
+    return {
+      awsAccessKeyId: process.env.AWS_ACCESS_KEY_ID,
+      awsSecretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    };
+  }
+
+  /**
+   * Publishes current lambda layer zip bundle to AWS Lambda.
+   * @param _version New version to be released.
+   * @param revision Git commit SHA to be published.
+   */
+  public async publish(_version: string, revision: string): Promise<any> {
+    logger.debug('Fetching artifact list...');
+    const packageFiles = await this.getArtifactsForRevision(revision, {
+      includeNames: DEFAULT_AWS_LAMBDA_DIST_REGEX,
+    });
+
+    if (packageFiles.length == 0) {
+      reportError('Cannot release to AWS Lambda: no packages found');
+      return undefined;
+    } else if (packageFiles.length > 1) {
+      reportError(`Cannot release to AWS Lambda:
+      multiple packages with matching patterns were found.`);
+      return undefined;
+    }
+
+    const artifactBuffer = fs.readFileSync(
+      await this.artifactProvider.downloadArtifact(packageFiles[0])
+    );
+
+    const publishRegionToLambda = async (currentRegion: string) => {
+      const lambda = new Lambda({ region: currentRegion });
+
+      const publishedLayer = await this.publishAwsLayer(lambda, {
+        Content: {
+          ZipFile: artifactBuffer,
+        },
+        LayerName: AwsLambdaTarget.layerName,
+        CompatibleRuntimes: compatibleRuntimes,
+        LicenseInfo: 'MIT',
+      });
+
+      if (publishedLayer.Version === undefined) {
+        reportError(`Error while publishing AWS Layer to ${currentRegion}`);
```

Comment:
s/AWS Layer/AWS Lambda Layer/

This error message is missing information to be useful, right now even reading the code leaves one puzzled regarding what happened.
Why do we check the `publishedLayer.Version`? Is it a proxy for the Lambda Client request worked fine? What should craft users (e.g. someone releasing sentry-javascript) do in case this error pops up? Re-run? Config error? Unknown?